File: android/app/src/main/java/com/guitaripod/pixie/data/api/model/ImageModels.kt
package com.guitaripod.pixie.data.api.model

import com.squareup.moshi.Json
import com.squareup.moshi.JsonClass

// Image Generation Models

@JsonClass(generateAdapter = true)
data class ImageGenerationRequest(
    @Json(name = "prompt") val prompt: String,
    @Json(name = "model") val model: String = "gpt-image-1",
    @Json(name = "n") val n: Int? = 1,
    @Json(name = "quality") val quality: String? = "auto",
    @Json(name = "size") val size: String? = "auto",
    @Json(name = "style") val style: String? = null,
    @Json(name = "response_format") val responseFormat: String? = "url",
    @Json(name = "user") val user: String? = null,
    @Json(name = "transparency") val transparency: String? = null,
    @Json(name = "background") val background: String? = null,
    @Json(name = "output_format") val outputFormat: String? = null,
    @Json(name = "output_quality") val outputQuality: Int? = null,
    @Json(name = "store") val store: Boolean? = true,
    @Json(name = "metadata") val metadata: Map<String, String>? = null
)

@JsonClass(generateAdapter = true)
data class ImageGenerationResponse(
    @Json(name = "created") val created: Long,
    @Json(name = "data") val data: List<ImageData>
)

@JsonClass(generateAdapter = true)
data class ImageData(
    @Json(name = "url") val url: String?,
    @Json(name = "b64_json") val b64Json: String?,
    @Json(name = "revised_prompt") val revisedPrompt: String?,
    @Json(name = "id") val id: String?,
    @Json(name = "metadata") val metadata: ImageMetadata?
)

@JsonClass(generateAdapter = true)
data class ImageMetadata(
    @Json(name = "width") val width: Int,
    @Json(name = "height") val height: Int,
    @Json(name = "format") val format: String,
    @Json(name = "size_bytes") val sizeBytes: Long,
    @Json(name = "credits_used") val creditsUsed: Int,
    @Json(name = "quality") val quality: String? = null,
    @Json(name = "model") val model: String? = null,
    @Json(name = "revised_prompt") val revisedPrompt: String? = null
)

// Gallery Models

@JsonClass(generateAdapter = true)
data class ImageListResponse(
    @Json(name = "images") val images: List<ImageDetails>,
    @Json(name = "total") val total: Int,
    @Json(name = "page") val page: Int,
    @Json(name = "per_page") val perPage: Int
)

@JsonClass(generateAdapter = true)
data class ImageDetails(
    @Json(name = "id") val id: String,
    @Json(name = "user_id") val userId: String,
    @Json(name = "url") val url: String,
    @Json(name = "thumbnail_url") val thumbnailUrl: String?,
    @Json(name = "prompt") val prompt: String,
    @Json(name = "created_at") val createdAt: String,
    @Json(name = "metadata") val metadata: ImageMetadata?,
    @Json(name = "is_public") val isPublic: Boolean? = null,
    @Json(name = "tags") val tags: List<String>?
)

File: iOS/Pixie/Pixie/Data/Models/ImageModels.swift
import Foundation

struct ImageGenerationRequest: Codable {
    let prompt: String
    let model: String
    let n: Int
    let size: String
    let quality: String
    let background: String?
    let moderation: String?
    let outputCompression: Int?
    let outputFormat: String?
    let partialImages: Int?
    let stream: Bool?
    let user: String?
    
    enum CodingKeys: String, CodingKey {
        case prompt, model, n, size, quality, background, moderation
        case outputCompression = "output_compression"
        case outputFormat = "output_format"
        case partialImages = "partial_images"
        case stream, user
    }
}

struct ImageEditRequest: Codable {
    let image: [String]
    let prompt: String
    let mask: String?
    let model: String
    let n: Int
    let size: String
    let quality: String
    let background: String
    let inputFidelity: String
    let outputFormat: String
    let outputCompression: Int?
    let partialImages: Int
    let stream: Bool
    let user: String?
    
    enum CodingKeys: String, CodingKey {
        case image, prompt, mask, model, n, size, quality, background
        case inputFidelity = "input_fidelity"
        case outputFormat = "output_format"
        case outputCompression = "output_compression"
        case partialImages = "partial_images"
        case stream, user
    }
}

struct ImageResponse: Codable {
    let created: TimeInterval
    let data: [ImageData]
}

struct ImageData: Codable {
    let url: String?
    let b64Json: String?
    let revisedPrompt: String?
    
    enum CodingKeys: String, CodingKey {
        case url
        case b64Json = "b64_json"
        case revisedPrompt = "revised_prompt"
    }
}

struct GalleryResponse: Codable {
    let images: [ImageMetadata]
    let total: Int
    let page: Int
    let perPage: Int
    
    enum CodingKeys: String, CodingKey {
        case images, total, page
        case perPage = "per_page"
    }
}

struct UserGalleryResponse: Codable {
    let userId: String
    let images: [ImageMetadata]
    let total: Int
    let page: Int
    let perPage: Int
    
    enum CodingKeys: String, CodingKey {
        case userId = "user_id"
        case images, total, page
        case perPage = "per_page"
    }
}

struct ImageMetadata: Codable, Hashable {
    let id: String
    let url: String
    let prompt: String
    let createdAt: String
    let userId: String
    let thumbnailUrl: String?
    let metadata: ImageMetadataDetails?
    let isPublic: Bool?
    let tags: [String]?
    
    enum CodingKeys: String, CodingKey {
        case id, url, prompt
        case createdAt = "created_at"
        case userId = "user_id"
        case thumbnailUrl = "thumbnail_url"
        case metadata
        case isPublic = "is_public"
        case tags
    }
}

struct ImageMetadataDetails: Codable, Hashable {
    let width: Int
    let height: Int
    let format: String
    let sizeBytes: Int
    let creditsUsed: Int
    let quality: String?
    let model: String?
    let revisedPrompt: String?
    
    enum CodingKeys: String, CodingKey {
        case width, height, format
        case sizeBytes = "size_bytes"
        case creditsUsed = "credits_used"
        case quality, model
        case revisedPrompt = "revised_prompt"
    }
}